restapi/src/main/java/io/stargate/web/docsapi/models/dto/CreateNamespace.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.web.docsapi.models.dto;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.stargate.db.query.builder.Replication;
+import io.swagger.annotations.ApiModelProperty;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.validation.Valid;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+
+/** The DTO for the namespace creation. */
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class CreateNamespace {
+
+  @JsonProperty("name")
+  @NotNull(message = "`name` is required to create a namespace")
+  @NotBlank(message = "`name` is required to create a namespace")
+  private final String name;
+
+  @JsonProperty("replicas")
+  @Min(value = 1, message = "minimum amount of `replicas` for `SimpleStrategy` is one")
+  private final Integer replicas;
+
+  @JsonProperty("datacenters")
+  @Valid
+  private final Collection<Datacenter> datacenters;
+
+  @JsonCreator
+  public CreateNamespace(
+      @JsonProperty("name") String name,
+      @JsonProperty("replicas") Integer replicas,
+      @JsonProperty("datacenters") Collection<Datacenter> datacenters) {
+    this.name = name;
+    this.replicas = replicas;
+    this.datacenters = datacenters;
+  }
+
+  @JsonIgnore
+  public Replication getReplication() {
+    if (null == datacenters || datacenters.isEmpty()) {
+      int replicas = Optional.ofNullable(this.replicas).orElse(1);
```

Comment:
Not a big deal, but could `replicas` and `datacenter` be optional types instead of converting them?

https://github.com/FasterXML/jackson-modules-java8

Reply:
Idea is good, I tried, but seems that validations are not working anymore for anything wrapped in the optional..